### PR TITLE
Permits TokenClassificationOption / DistilBertForTokenClassification to fail gracefully for an invalid configuration.

### PR DESCRIPTION
**Pull request by andyHa**
A production system shouldn't panic, if an invalid config is present. The implementation
is placed in a new method, so that the change is non-breaking.

### CODE REVIEW

File: src/distilbert/distilbert_model.rs [hunk 1]
```diff
@@ -682,30 +682,59 @@ impl DistilBertForTokenClassification {
...
-            .len() as i64;
-
-        let classifier = nn::linear(p / "classifier", config.dim, num_labels, Default::default());
-        let dropout = Dropout::new(config.seq_classif_dropout);
-
-        DistilBertForTokenClassification {
-            distil_bert_model,
-            classifier,
-            dropout,
+        if let Some(id2label) = &config.id2label {
```
Comment (guillaume-be) [inline src/distilbert/distilbert_model.rs:693]: Could raising the error be closer to the problematic field using 
```rust
        let num_labels = config
            .id2label
            .as_ref()
            .ok_or_else(|| RustBertError::InvalidConfigurationError("id2label must be provided in the configuration for sequence classification".to_string()))?
            .len() as i64;
```
This also makes this code block more generic and reusable across the crate

File: src/distilbert/distilbert_model.rs [hunk 2]
```diff
@@ -682,30 +682,59 @@ impl DistilBertForTokenClassification {
     /// let config = DistilBertConfig::from_file(config_path);
     /// let distil_bert = DistilBertForTokenClassification::new(&p.root() / "distilbert", &config);
     /// ```
-    pub fn new<'p, P>(p: P, config: &DistilBertConfig) -> DistilBertForTokenClassification
-    where
-        P: Borrow<nn::Path<'p>>,
+    pub fn try_new<'p, P>(p: P, config: &DistilBertConfig) -> Result<DistilBertForTokenClassification, RustBertError>
+        where
```
Comment (guillaume-be) [inline src/distilbert/distilbert_model.rs:686]: These edits do not appear when I run `rustfmt`, could you please see if you are using the latest binaries with default settings?

File: src/distilbert/distilbert_model.rs [hunk 3]
```diff
@@ -682,30 +682,59 @@ impl DistilBertForTokenClassification {
     /// let config = DistilBertConfig::from_file(config_path);
     /// let distil_bert = DistilBertForTokenClassification::new(&p.root() / "distilbert", &config);
     /// ```
-    pub fn new<'p, P>(p: P, config: &DistilBertConfig) -> DistilBertForTokenClassification
-    where
-        P: Borrow<nn::Path<'p>>,
+    pub fn try_new<'p, P>(p: P, config: &DistilBertConfig) -> Result<DistilBertForTokenClassification, RustBertError>
```
Comment (guillaume-be) [inline src/distilbert/distilbert_model.rs:685]: The API is still undergoing regular re-designs and breaking changes will be pushed for the next release in a few weeks. I would be fine introducing a consistent `new` method for the `SequenceClassificationModels` that would be breaking and avoid adding a `try_new` method to try and keep the API uncluttered. What do you think?

**Review by guillaume-be — COMMENTED** [review]
Thank you for the suggestion @andyHa ,
I have a few minor comments on the DistilBert implementation. This applies to other models, could you please update all instances of `unwrapping/expecting` in these model instantiations for consistency ? If you do not have the capacity please let me know I'd be happy to support.

File: src/distilbert/distilbert_model.rs (hunk 1 above)
Comment (andyHa) [reply src/distilbert/distilbert_model.rs]: Thank's fror the hint - didn't think of that +1

File: src/distilbert/distilbert_model.rs (hunk 2 above)
Comment (andyHa) [reply src/distilbert/distilbert_model.rs]: Somehow intellij / cargo tricked me here :-/ ...

File: src/distilbert/distilbert_model.rs (hunk 3 above)
Comment (andyHa) [reply src/distilbert/distilbert_model.rs]: Great idea - done.

**Comment by andyHa** [comment]
> Thank you for the suggestion @andyHa , I have a few minor comments on the DistilBert implementation. This applies to other models, could you please update all instances of `unwrapping/expecting` in these model instantiations for consistency ? If you do not have the capacity please let me know I'd be happy to support.

@guillaume-be thanks for the feedback - I updated all methods to consistently return a Result and changed all .expect to .ok_or_else(..)?

File: src/fnet/fnet_model.rs [hunk 4]
```diff
@@ -958,7 +965,7 @@ impl FNetForQuestionAnswering {
     /// # let device = Device::Cpu;
     /// # let vs = nn::VarStore::new(device);
     /// # let config = FNetConfig::from_file(config_path);
-    /// let model = FNetForTokenClassification::new(&vs.root(), &config);
+    /// let model = FNetForTokenClassification::new(&vs.root(), &config).unwrap();
```
Comment (guillaume-be) [inline src/fnet/fnet_model.rs:968]: it looks like this docstring was wrong to start with - could you please fix it with the question answering model instead?

Comment (andyHa) [reply src/fnet/fnet_model.rs]: Done :)

File: tests/fnet.rs [hunk 5]
```diff
@@ -227,7 +228,7 @@ fn fnet_for_token_classification() -> anyhow::Result<()> {
 
     //    Forward pass
     let model_output = no_grad(|| {
-        fnet_model
+        fnet_model.unwrap()
```
Comment (guillaume-be) [inline tests/fnet.rs:231]: can you please be consistent for all tests with the model error handling  (see the changes for `albert` for example):
`let albert_model = AlbertForTokenClassification::new(vs.root(), &config)?;`

Comment (andyHa) [reply tests/fnet.rs]: Done

**Pull request merged**